### PR TITLE
fix udev DMI rules – make sure hostnamectl shows correct hw info even after device trigger

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/hostname/hostnamed.c [hunk 1]
```diff
@@ -185,6 +185,10 @@ static int get_dmi_data(const char *database_key, const char *regular_key, char
         const char *s = NULL;
         int r;
 
+        if (detect_container() > 0)
+                return log_debug_errno(SYNTHETIC_ERRNO(ENOENT),
+                                       "Running in a container, suppressing DMI hardware data.");
```
Comment (bluca) [inline src/hostname/hostnamed.c]: "suppressing" sounds weird, maybe "skipping" or "ignoring"?

File: rules.d/meson.build [hunk 2]
```diff
@@ -8,6 +8,7 @@ rules = [
         [files('60-autosuspend.rules',
                '60-block.rules',
                '60-cdrom_id.rules',
+               '60-dmi-id.rules',
```
Comment (bluca) [inline rules.d/meson.build:11]: this should be in the first commit rather than the last?

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
```
21:50:16 [ 1166.855176] testsuite-71.sh[125]: + assert_in 2000-01-01 ' Static hostname: H
21:50:16 [ 1166.855236] testsuite-71.sh[125]:        Icon name: computer-container
21:50:16 [ 1166.855265] testsuite-71.sh[125]:          Chassis: container ☐
21:50:16 [ 1166.855297] testsuite-71.sh[125]:       Machine ID: f634b29277374c6997898f931f72d937
21:50:16 [ 1166.855330] testsuite-71.sh[125]:          Boot ID: f87aaf4630bc4bcca1e2244563c5357b
21:50:16 [ 1166.855363] testsuite-71.sh[125]:   Virtualization: systemd-nspawn
21:50:16 [ 1166.855395] testsuite-71.sh[125]: Operating System: Arch Linux
21:50:16 [ 1166.855430] testsuite-71.sh[125]:           Kernel: Linux 6.3.8-arch1-1
21:50:16 [ 1166.855466] testsuite-71.sh[125]:     Architecture: x86-64'
21:50:16 [ 1166.855883] testsuite-71.sh[156]: + set +ex
21:50:16 [ 1166.855938] testsuite-71.sh[156]: FAIL: '2000-01-01' not found in:
21:50:16 [ 1166.855977] testsuite-71.sh[156]:  Static hostname: H
21:50:16 [ 1166.856013] testsuite-71.sh[156]:        Icon name: computer-container
21:50:16 [ 1166.856052] testsuite-71.sh[156]:          Chassis: container ☐
21:50:16 [ 1166.856098] testsuite-71.sh[156]:       Machine ID: f634b29277374c6997898f931f72d937
21:50:16 [ 1166.856133] testsuite-71.sh[156]:          Boot ID: f87aaf4630bc4bcca1e2244563c5357b
21:50:16 [ 1166.856167] testsuite-71.sh[156]:   Virtualization: systemd-nspawn
21:50:16 [ 1166.856200] testsuite-71.sh[156]: Operating System: Arch Linux
21:50:16 [ 1166.856234] testsuite-71.sh[156]:           Kernel: Linux 6.3.8-arch1-1
21:50:16 [ 1166.856267] testsuite-71.sh[156]:     Architecture: x86-64
```

**Comment by poettering** [comment]
CI hopefully fixed now.

i added one more commit on top. hence ptal, again.

**Pull request merged**